src/GHCMod/Options/Commands.hs

```diff
@@ -0,0 +1,230 @@
+module GHCMod.Options.Commands where
+
+import Options.Applicative
+import Options.Applicative.Types
+import Language.Haskell.GhcMod.Types
+import GHCMod.Options.DocUtils
+import Language.Haskell.GhcMod.Read
+
+type Symbol = String
+type Expr = String
+type Module = String
+type Line = Int
+type Col = Int
+type Point = (Line, Col)
+
+data GhcModCommands =
```

Comment:
It does have the benefit of making integration via the API for e.g. HIE more type-safe.